src/errors.rs

```
@@ -13,6 +13,20 @@
 //! }
 //! ```
 
+use std::fmt::{Display, Formatter};
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum RuntimeErrorCode {
+    RemoteServiceUnavailable,
+    GenericError,
```

Not sure whether I like it to propose that one could use an Error that doesn't say anything about its nature. Why is this required?

---

We might want to throw an error which we know is a RuntimeError, but we don't have a code for it.

---

For example, [syncing the bdk wallet](https://docs.rs/bdk/0.25.0/bdk/wallet/struct.Wallet.html#method.sync) can return a [generic BDK error](https://docs.rs/bdk/0.25.0/bdk/enum.Error.html#variant.Generic). What should we do in that case?

---

Yeah, well in that case we can only forward the inaccurateness, I agree.

Could we at least have a string to add more detail to the error, similar to what the generic BDK error does, or is that not possible because of how our error handling works (resp. because of UNIFFI binding limitations)?

---

When throwing this error code we still have the `message` field of in the `RuntimeError` so I think we don't need to add further details as part of the code itself.

---

Oh okay, thanks for elaborating!